### PR TITLE
5 implement weather api

### PR DESCRIPTION
**Pull request by GrubeCube**
The weather API is complete and ready for use when GUI is completed. The weather API retrieves forecast information using the National Weather Service's API. The API allows two major methods for retrieving information:

- **updateForecast():** requests updated information from the NWS API using an asynchronous Volley task. Sends a local broadcast to alert UI thread when it is okay to access data.
- **getForecast():** returns an array of all forecast information separated by time.

In addition, the weather forecast can search through the parsed forecast for keywords that would indicate unfavorable weather conditions.

### CODE REVIEW

File: app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+                    mForecast = tmpForecast;
+
+                    Intent requestComplete = new Intent(WEATHER_ACTION);
+                    LocalBroadcastManager.getInstance(mAppContext).sendBroadcast(requestComplete);
+                } catch (JSONException e) {
+                    Log.w(LOG_TAG, "Error parsing weather forecast: " + e.getMessage());
+                }
+            }
+        };
+        Response.ErrorListener reqErrListener = new Response.ErrorListener() {
```
Comment (nbice) [inline app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java]: Add newline.

File: app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+         */
+        private String mForecast, mWind, mTemperature;
+
+        /**
+         * JSON parsing constructor primarily used for parsing the reponse from
+         * NWS API.
+         *
+         * @throws JSONException    invalid or unexpected formatted JSON
+         */
+        public WeatherForecast(JSONObject forecast) throws JSONException {
```
Comment (nbice) [inline app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java]: Add @param JSONObject forecast?

File: app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+
+        public String getForecast() {
+            return mForecast;
+        }
+
+        public String getWind() {
+            return mWind;
+        }
+
+        public String geTemperature() {
```
Comment (nbice) [inline app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java]: *getTemperature

File: app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+                }
+            }
+        };
+        Response.ErrorListener reqErrListener = new Response.ErrorListener() {
+            @Override
+            public void onErrorResponse(VolleyError e) {
+                Log.w(LOG_TAG, "Could not retrieve weather forecast: " + e.getMessage());
+            }
+        };
+        NwsForecastRequest request = new NwsForecastRequest(reqUrl, reqListener, reqErrListener);
```
Comment (nbice) [inline app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java]: Add newline.

File: app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+
+    /**
+     * Time the last successful update occured.
+     */
+    private Instant mLastUpdate;
+
+    /**
+     * Forecast data of reponse.
+     */
+    private WeatherForecast[] mForecast;
```
Comment (nbice) [inline app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java]: Is this plural or singular?

File: app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java [hunk 6]
```diff
@@ -0,0 +1,354 @@
...
+
+
+/**
+ * Retrieves forecast information using the National Weather Service API. The
+ * returned information can be checked for unfavorable conditions or retrieved
+ * using the <code>WeatherForecast</code> inner class.
+ *
+ * @author      Thomas Gruber
+ * @version     0.9, 09/22/18
+  *
```
Comment (nbice) [inline app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java]: Format, remove space.

File: app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java (hunk 1 above)
Comment (GrubeCube) [reply app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java]: Commit 35a5a7d829e7045a76db3fae4318fde20751550d has added the newline.

File: app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java (hunk 2 above)
Comment (GrubeCube) [reply app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java]: Commit 35a5a7d829e7045a76db3fae4318fde20751550d has added the @param comment.

File: app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java (hunk 3 above)
Comment (GrubeCube) [reply app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java]: Commit 35a5a7d829e7045a76db3fae4318fde20751550d resolved the typo. There were no references to this function.

File: app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java (hunk 4 above)
Comment (GrubeCube) [reply app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java]: Commit 35a5a7d829e7045a76db3fae4318fde20751550d added the newline.

File: app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java (hunk 6 above)
Comment (GrubeCube) [reply app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java]: Commit 35a5a7d829e7045a76db3fae4318fde20751550d removed the extra space.

File: app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java (hunk 5 above)
Comment (GrubeCube) [reply app/src/main/java/com/terradrone/ceg4981/WeatherRetriever.java]: Commit 35a5a7d829e7045a76db3fae4318fde20751550d updated the function to use the plural form (i.e. getForecasts()) and updated any references to the function. This was also important to differentiate the function to the nested class function of the same name.

**Pull request merged**